core/src/test/java/org/elasticsearch/index/query/QueryParseContextTests.java

```diff
@@ -64,40 +65,11 @@ public void teardown() throws IOException {
         this.threadContext.close();
     }
 
-    public void testParseTopLevelBuilder() throws IOException {
-        QueryBuilder query = new MatchQueryBuilder("foo", "bar");
-        String requestBody = "{ \"query\" : " + query.toString() + "}";
-        try (XContentParser parser = createParser(JsonXContent.jsonXContent, requestBody)) {
-            QueryParseContext context = new QueryParseContext(parser);
-            QueryBuilder actual = context.parseTopLevelQueryBuilder();
-            assertEquals(query, actual);
-        }
-    }
-
-    public void testParseTopLevelBuilderEmptyObject() throws IOException {
-        String requestBody = "{}";
-        try (XContentParser parser = createParser(JsonXContent.jsonXContent, requestBody)) {
-            QueryParseContext context = new QueryParseContext(parser);
-            QueryBuilder query = context.parseTopLevelQueryBuilder();
-            assertNull(query);
-        }
-    }
-
-    public void testParseTopLevelBuilderUnknownParameter() throws IOException {
-        String requestBody = "{ \"foo\" : \"bar\"}";
-        try (XContentParser parser = createParser(JsonXContent.jsonXContent, requestBody)) {
-            QueryParseContext context = new QueryParseContext(parser);
-            ParsingException exception = expectThrows(ParsingException.class, () ->  context.parseTopLevelQueryBuilder());
-            assertEquals("request does not support [foo]", exception.getMessage());
-        }
-    }
-
     public void testParseInnerQueryBuilder() throws IOException {
```

Comment:
Already part of a follow-up PR. I can do it now if you prefer, but the test is split there anyway.